src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -1685,6 +1685,11 @@ protected override void EnterParameter(ParameterSymbol parameter)
                 if (slot > 0) SetSlotState(slot, true);
                 NoteWrite(parameter, value: null, read: true);
             }
+
+            if (parameter.BoundEqualsValue is { } boundValue)
+            {
+                VisitRvalue(boundValue.Value);
```

Comment:
>VisitRvalue(boundValue.Value); [](http://example.com/codeflow?start=16&length=30)

Do we want to limit this to lambda parameters only?

Reply:
And parameters of local functions too, I guess.

Reply:
Ok! Otherwise, we would be visiting default parameter values for methods too. But it looks like we don't do unused field warnings for constants from what I can tell from the code [here](https://github.com/dotnet/roslyn/blob/3e39dd3535962bf9e30bd650e4ff34b610b8349a/src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs#L697), correct? So visiting the default parameter expression for a method would be useless work?